### ISSUE TITLE
babel-cli postinstall spam

### ISSUE THREAD

**Issue by tildechris** [babel/babel#10152]
## Bug Report

**Current Behavior**
When installing babel-cli, a transitive dependency is spamming the installation log.  Consider removing `core-js` as a dependency.

```
$ npm install
...
> core-js@2.6.9 postinstall /usr/local/google/home/tildechris/src/tildechris-expeditors/node_modules/core-js
> node scripts/postinstall || echo "ignore"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)


> core-js-pure@3.1.4 postinstall /usr/local/google/home/tildechris/src/tildechris-expeditors/node_modules/core-js-pure
> node scripts/postinstall || echo "ignore"

Thank you for using core-js ( https://github.com/zloirock/core-js ) for polyfilling JavaScript standard library!

The project needs your help! Please consider supporting of core-js on Open Collective or Patreon: 
> https://opencollective.com/core-js 
> https://www.patreon.com/zloirock 

Also, the author of core-js ( https://github.com/zloirock ) is looking for a good job -)
```

**Expected behavior/code**

Installation without error should not produce any output by default.

**Environment**
- Babel version(s): [e.g. 6.26.0]
- Node/npm version: [e.g. node v11.10.0/npm 6.7.0]
- OS: [Debian GNU/Linux rodete]
- Monorepo: [e.g. no]
- How you are using Babel: [`cli`]

**Possible Solution**
Consider removing `core-js` as a dependency.

**Additional context/Screenshots**

Please see below for more context:

https://github.com/zloirock/core-js/issues/548

tl;dr - The core-js maintainer isn't willing to revert this regression.

**Comment by zloirock** [comment]
Good luck -)

**Comment by tildechris** [comment]
Thanks @zloirock.  Hopefully this request remains civil.  Your package obviously delivers value, but IMHO this is a regression.

Please let me know if there's anything that would help you reconsider your decision.

**Comment by waynebloss** [comment]
@tildechris Don't even worry about it - the people who want to push ads in your face always get spanked and this will be no different.

People clearly don't like ads in general, never mind this desperate, pathetic and distasteful personal ad inappropriately spewed into your logs.  Just look at the amount of people running adblock who disrupted Google and the rest of the big-money ad industry.

You might consider even encouraging this bad behavior so we can reach endgame more quickly.

If that's not enough for you right now, I would also encourage you and anyone else who cares to fill `core-js` issues with their own personal ads while also complaining directly to NPM.

**Comment by tildechris** [comment]
Thanks @waynebloss.  In order to evaluate this issue effectively, we should keep commentary on the upsteam issue at zloirock/core-js#548.  

Let's limit this thread to its effect on babel-cli and possible solutions.

**Comment by waynebloss** [comment]
@tildechris Yeah no problem. You won't see this particular account commenting there because apparently @zloirock can dish it out, but can't take it when people post inappropriate messages in inappropriate places that take up his or her resources.

The possible solutions that I can see:

- Complain to NPM until they kick all the bad actors off their platform.
- Complain to every package that uses this junk.
- Find someone to fork and maintain it without ads.
- Wait until the bad actors wake up and smell their own feces.

Eventually something enough things will break and we'll win. There's no doubt in my mind.

**Comment by jhpratt** [comment]
core-js is a major dependency, as it's used for a few plugins. Frankly, there's not a chance it's going to be removed.

**Comment by nicolo-ribaudo** [comment]
I can't reproduce this report: when I run `npm install --save-dev @babel/cli` I don't get the postinstall message.

I only get it from `@babel/preset-env`. I have talked with other people from the team, and we have decided to hide that message for users who **don't use Babel to inject `core-js`** in their code. If you use Babel to inject imports to `core-js` (or to `@babel/runtime-corejs3`), you'll still see the postinstall message coming from one of those two packages, because you are directly depending on `core-js`.

**Comment by tildechris** [comment]
Thanks @nicolo-ribaudo.  I should've added the only reason I thought it was babel-cli was because of

```
$ npm ls core-js
└─┬ babel-cli@6.26.0
  ├─┬ babel-polyfill@6.26.0
  │ └── core-js@2.6.9 
  ├─┬ babel-register@6.26.0
  │ └── core-js@2.6.9  deduped
  └─┬ babel-runtime@6.26.0
    └── core-js@2.6.9  deduped
```

I agree that it's a reasonable compromise to only show this message if you directly depend on core-js.

**Comment by nicolo-ribaudo** [comment]
Fixed by https://github.com/zloirock/core-js/pull/590

Current status:
1. Babel 6 packages (@tildechris, used by you) aren't maintained anymore, except maybe for highly problematic securety issues. I don't know how they behave.
1. `@babel/preset-env` and `@babel/plugin-transform-runtime` **don't show** the message when installed, since who uses those packages might not use `core-js`. If they use `core-js` using one of those package, they will see the message because of other dependencies.
1. `@babel/runtime-corejs2`, `@babel/runtime-corejs3` and `@babel/polyfill` **do show** the postinstall message, snce they are just wrappers around `core-js`. We won't remove it (and probably we can't), since it's a decision which can only be taken by the `core-js` author.
1. `@babel/node` **does show** the message, since internally it loads `core-js` before running your code to make it work in older node versions.
